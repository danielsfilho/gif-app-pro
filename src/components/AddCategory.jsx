import { useState } from "react"

export const AddCategory = ( { onNewCategory } ) => {
  const [inputValue, setInputValue] = useState('')

  const onInputChange = ( e ) => {
    setInputValue(e)
  }

  const onSubmit = ( event ) => {
    event.preventDefault()

    const newInputValue = inputValue.trim()
    if( newInputValue.length <= 0 ) return;

    onNewCategory( newInputValue  )
    setInputValue("")
  }

  return (
    <form onSubmit={ onSubmit }>
      
      <input
        type="text"
        placeholder="Procurar gifs"
        value={ inputValue }
        onChange={e => onInputChange(e.target.value)}
      />

    </form>
  )
}
