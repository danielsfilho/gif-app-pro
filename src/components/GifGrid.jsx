import { useFetchGifs } from "../hooks"
import { GifItem } from "./index"

export const GifGrid = ({ category }) => {

  const { images, isLoading } = useFetchGifs( category )

  return (
    <>
      <h3>{ category }</h3>

      {
        isLoading && ( <h1>Carregando...</h1> )
      }

      <div className="card-grid">
        { 
          images.map( ( image ) => (
            <GifItem 
              key={ image.id } 
              { ...image }
            />
          ))
        }
      </div>
    </>
  )
}
