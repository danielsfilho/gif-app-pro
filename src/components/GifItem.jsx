
export const GifItem = ({ title, url, id }) => {
  return (
    <div className="card">
      <img src={ url } alt={title} />

      <h1>
        <li>{ title }</li>
      </h1>
    </div>
  )
}
