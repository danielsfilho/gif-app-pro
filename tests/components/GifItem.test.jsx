import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";


describe('Provas no <GifItem />', () => { 

  const title = 'Saitama';
  const url = 'https://one-punch.com/saitama.jpg';

  test('deve fazer match com o snapshot', () => { 
    
    const { container } = render( <GifItem title={ title } url={ url } /> );
    expect( container ).toMatchSnapshot();
  });

  test('deve mostrar a imagem com a URL e o ALT indicado', () => { 
    
    render( <GifItem title={ title } url={ url } /> );
    // screen.debug();
    // expect( screen.getByRole('img').src ).toBe( url );
    // expect( screen.getByRole('img').alt ).toBe( title );
    const { src, alt } = screen.getByRole('img');
    expect( src ).toBe( url );
    expect( alt ).toBe( title );

  });

  test('deve mostrar o titulo no componente', () => { 
    
    render( <GifItem title={ title } url={ url } /> );
    
    expect( screen.getByText( title ) ).toBeTruthy()

  }); 
});