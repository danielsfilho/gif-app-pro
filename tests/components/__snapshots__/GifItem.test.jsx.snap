// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Provas no <GifItem /> deve fazer match com o snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <h1>
      <li>
        Saitama
      </li>
    </h1>
  </div>
</div>
`;
